code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -13,481 +13,374 @@ What are the archived variables for?
                                                             Edit: As far as I know this might have a bug caused by round(). When it has a second arg it will round up.
                                                             So for instance round(0.5, 1) == 1. Trouble is I haven't found any instances of it causing a bug,
                                                             and any attempts to fix it just killed atmos. I leave this to a greater man then I*/
-GLOBAL_LIST_INIT(meta_gas_info, meta_gas_list()) //see ATMOSPHERICS/gas_types.dm
-GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
-
-/proc/init_gaslist_cache()
-	. = list()
-	for(var/id in GLOB.meta_gas_info)
-		var/list/cached_gas = new(3)
-
-		.[id] = cached_gas
-
-		cached_gas[MOLES] = 0
-		cached_gas[ARCHIVE] = 0
-		cached_gas[GAS_META] = GLOB.meta_gas_info[id]
+//Unomos - global list inits for all of the meta gas lists.
+//This setup allows procs to only look at one list instead of trying to dig around in lists-within-lists
+GLOBAL_LIST_INIT(meta_gas_specific_heats, meta_gas_heat_list())
+GLOBAL_LIST_INIT(meta_gas_names, meta_gas_name_list())
+GLOBAL_LIST_INIT(meta_gas_visibility, meta_gas_visibility_list())
+GLOBAL_LIST_INIT(meta_gas_overlays, meta_gas_overlay_list())
+GLOBAL_LIST_INIT(meta_gas_dangers, meta_gas_danger_list())
+GLOBAL_LIST_INIT(meta_gas_ids, meta_gas_id_list())
+GLOBAL_LIST_INIT(meta_gas_fusions, meta_gas_fusion_list())
 
 /datum/gas_mixture
-	var/list/gases
-	var/temperature = 0 //kelvins
-	var/tmp/temperature_archived = 0
-	var/volume = CELL_VOLUME //liters
-	var/last_share = 0
+	/// Do not directly set this var. Externally hooked by extools, used for inspection in VV
+	var/gas_list_view_only
+	var/initial_volume = CELL_VOLUME
 	var/list/reaction_results
 	var/list/analyzer_results //used for analyzer feedback - not initialized until its used
-	var/gc_share = FALSE // Whether to call garbage_collect() on the sharer during shares, used for immutable mixtures
+	/// Contains the memory address of the shared_ptr object for this gas mixture in c++ land. Don't. Touch. This. Var.
+	var/_extools_pointer_gasmixture = 0
 
 /datum/gas_mixture/New(volume)
-	gases = new
 	if (!isnull(volume))
-		src.volume = volume
+		initial_volume = volume
```

Comment:
Why this change?

Reply:
`volume` doesn't exist on gas_mixture anymore. `initial_volume` is read by extools to setup the gas mixture on its end.